src/vs/workbench/services/issue/electron-sandbox/issueService.ts

```diff
@@ -82,6 +87,22 @@ export class NativeIssueService implements IWorkbenchIssueService {
 			const result = [this._providers.has(extensionId.toLowerCase()), this._handlers.has(extensionId.toLowerCase())];
 			ipcRenderer.send('vscode:triggerReporterStatusResponse', result);
 		});
+		ipcRenderer.on('vscode:triggerReporterMenu', async (event, arg) => {
+			const extensionId = arg.extensionId;
+
+			// creates menu from contributed
+			const menu = this.menuService.createMenu(MenuId.IssueReporter, this.contextKeyService);
+
+			// render menu and dispose
+			const actions = menu.getActions({ renderShortTitle: true }).flatMap(entry => entry[1]);
+			actions.forEach(async action => {
+				if (action.item && 'source' in action.item && action.item.source?.id === extensionId) {
+					this.foundExtension = true;
```

Comment:
This seems a bit fragile.

* What happens if the action throws?
* What happens if it never calls the command again?
* What happens if a different extension calls the command while waiting for the action to respond?

Setting a property on the service could lead to a bad state.

Reply:
added fixes 👍